### ISSUE TITLE
How to read a classification model to do classification

### ISSUE THREAD

**Issue by qianqing13579** [ROCm/AMDMIGraphX#601]
I am learning migraphx recently. But I do not know how to read a classification model(such as resnet50) to do classification  by using migraphx.At the same time,I also want to know how to use SSD detector by migraphx

**Comment by pfultz2** [comment]
Here's a simple example that shows how to classify an image, but the `load_image` function is not shown:

```python
import numpy as np
import migraphx

# Load the resnet50 onnx file
model = migraphx.parse_onnx("resnet50.onnx")
# Compile the model for the gpu
model.compile(migraphx.get_target("gpu"))

# Load image as a numpy array
image = load_image()

# Run the model, and pass the image as parameter '0'
result = model.run({'0': migraphx.argument(image)})

# Convert to numpy array
npresult = np.array(model.run(params), copy=False)

# Image classification index
idx = np.argmax(npresult[0])
```

Here is an end-to-end example of classifying images with a webcam. This requires python-opencv and a json file for the labels:

```python
import numpy as np
import cv2
import json
import migraphx

# video settings
cap = cv2.VideoCapture(0)
cap.set(cv2.cv.CV_CAP_PROP_FRAME_WIDTH, 320)
cap.set(cv2.cv.CV_CAP_PROP_FRAME_HEIGHT, 240)
ret, frame = cap.read()

# neural network settings
model = migraphx.parse_onnx("resnet50.onnx")
model.compile(migraphx.get_target("gpu"))

# get labels
with open('imagenet_class_index.json') as json_data:
    class_idx = json.load(json_data)
    idx2label = [class_idx[str(k)][1] for k in range(len(class_idx))]

# primary loop to read webcam images
count = 0
while (True):
    # capture frame by frame
    ret, frame = cap.read()

    if ret: # check - some webcams need warmup operations on the frame
        cropped = frame[16:304,8:232]    # 224x224

        trans = cropped.transpose(2,0,1) # convert HWC to CHW

        # convert to float, normalize and make batch size = 1
        image = np.ascontiguousarray(
            np.expand_dims(trans.astype('float32')/256.0,0))

        # display the frame
        cv2.imshow('frame',cropped)

        params['0'] = migraphx.argument(image)

        result = np.array(model.run(params),copy=False)

        idx = np.argmax(result[0])

        print idx2label[idx], " ", result[0][idx]

    if cv2.waitKey(1) & 0xFF == ord('q'):
        break

# when everything is done, release the capture
cap.release()
cv2.destroyAllWindows()
```

**Comment by qianqing13579** [comment]
> Here's a simple example that shows how to classify an image, but the `load_image` function is not shown:
> 
> ```python
> import numpy as np
> import migraphx
> 
> # Load the resnet50 onnx file
> model = migraphx.parse_onnx("resnet50.onnx")
> # Compile the model for the gpu
> model.compile(migraphx.get_target("gpu"))
> 
> # Load image as a numpy array
> image = load_image()
> 
> # Run the model, and pass the image as parameter '0'
> result = model.run({'0': migraphx.argument(image)})
> 
> # Convert to numpy array
> npresult = np.array(model.run(params), copy=False)
> 
> # Image classification index
> idx = np.argmax(npresult[0])
> ```
> 
> Here is an end-to-end example of classifying images with a webcam. This requires python-opencv and a json file for the labels:
> 
> ```python
> import numpy as np
> import cv2
> import json
> import migraphx
> 
> # video settings
> cap = cv2.VideoCapture(0)
> cap.set(cv2.cv.CV_CAP_PROP_FRAME_WIDTH, 320)
> cap.set(cv2.cv.CV_CAP_PROP_FRAME_HEIGHT, 240)
> ret, frame = cap.read()
> 
> # neural network settings
> model = migraphx.parse_onnx("resnet50.onnx")
> model.compile(migraphx.get_target("gpu"))
> 
> # get labels
> with open('imagenet_class_index.json') as json_data:
>     class_idx = json.load(json_data)
>     idx2label = [class_idx[str(k)][1] for k in range(len(class_idx))]
> 
> # primary loop to read webcam images
> count = 0
> while (True):
>     # capture frame by frame
>     ret, frame = cap.read()
> 
>     if ret: # check - some webcams need warmup operations on the frame
>         cropped = frame[16:304,8:232]    # 224x224
> 
>         trans = cropped.transpose(2,0,1) # convert HWC to CHW
> 
>         # convert to float, normalize and make batch size = 1
>         image = np.ascontiguousarray(
>             np.expand_dims(trans.astype('float32')/256.0,0))
> 
>         # display the frame
>         cv2.imshow('frame',cropped)
> 
>         params['0'] = migraphx.argument(image)
> 
>         result = np.array(model.run(params),copy=False)
> 
>         idx = np.argmax(result[0])
> 
>         print idx2label[idx], " ", result[0][idx]
> 
>     if cv2.waitKey(1) & 0xFF == ord('q'):
>         break
> 
> # when everything is done, release the capture
> cap.release()
> cv2.destroyAllWindows()
> ```

Do you know how to use SSD detector by migraphx?

**Comment by kahmed10** [comment]
@qianqing13579 do you have a specific SSD model you are testing?

**Comment by qianqing13579** [comment]
Here is the link: https://github.com/qianqing13579/SafetyHelmet/tree/master/Resource/Models/Detector. It is a caffemodel

**Comment by kahmed10** [comment]
> Here is the link: https://github.com/qianqing13579/SafetyHelmet/tree/master/Resource/Models/Detector. It is a caffemodel

MIGraphX currently does not support caffe models. Can you try converting to onnx? Depending on the SSD model implementation, MIGraphX may be able to support it.

**Comment by qianqing13579** [comment]
> > Here is the link: https://github.com/qianqing13579/SafetyHelmet/tree/master/Resource/Models/Detector. It is a caffemodel
> 
> MIGraphX currently does not support caffe models. Can you try converting to onnx? Depending on the SSD model implementation, MIGraphX may be able to support it.

Here is the link: https://github.com/qianqing13579/MIGraphX_SSD_Model

**Comment by qianqing13579** [comment]
> > Here is the link: https://github.com/qianqing13579/SafetyHelmet/tree/master/Resource/Models/Detector. It is a caffemodel
> 
> MIGraphX currently does not support caffe models. Can you try converting to onnx? Depending on the SSD model implementation, MIGraphX may be able to support it.

Here is the link: https://github.com/qianqing13579/MIGraphX_SSD_Model. I do not know how to use this SSD model by MIGraphX

**Comment by kahmed10** [comment]
There are onnx operators in this model that MIGraphX doesn't support natively. We have opened issues (#611, #612) to track their development. Once complete, we can provide additional support to help run this model using MIGraphX.

**Comment by scxiao** [comment]
@qianqing13579 Can you reopen the link then we can access the model? Looks like the above link is invalid now.

**Comment by qianqing13579** [comment]
@kahmed10 ,I have developed the SSD detector inference code by MIGraphX. Here is the link:https://github.com/qianqing13579/MIGraphX_Samples. Recently I have been developing the YOLOV3  inference code.
